client/blindai/client.py

```diff
@@ -439,19 +443,17 @@ def connect_server(
     def upload_model(
         self,
         model: str,
-        shape: Tuple = None,
-        dtype: ModelDatumType = ModelDatumType.F32,
-        dtype_out: ModelDatumType = ModelDatumType.F32,
+        tensor_inputs: Union[List[Any], List[List]],
+        tensor_outputs: Union[ModelDatumType, List[ModelDatumType]],
         sign: bool = False,
     ) -> UploadModelResponse:
         """Upload an inference model to the server.
         The provided model needs to be in the Onnx format.
 
         Args:
             model (str): Path to Onnx model file.
-            shape (Tuple, optional): The shape of the model input. Defaults to None.
-            dtype (ModelDatumType, optional): The type of the model input data (f32 by default). Defaults to ModelDatumType.F32.
-            dtype_out (ModelDatumType, optional): The type of the model output data (f32 by default). Defaults to ModelDatumType.F32.
+            tensor_inputs (Union[List[Any], List[List]]): A list describing multiple inputs of the model.
```

Comment:
Is it possible to just pass a tensor size and a dtype in case we have a model with one input, like in the current API?

Reply:
Okay, I see what you mean.
It is possible but obviously, there will have to be a little type checking before submitting the request to the server.
Will start working on it as well.